src/app/lib/earn.ts

```diff
@@ -0,0 +1,6 @@
+const API_URL = 'https://bitcoinfees.earn.com/api/v1/fees/recommended';
```

Comment:
To make this decision, I searched high and low trying to find ways to estimate fees. Without direct access to the mempool, we cannot do it ourselves. I found that [Bitgo](https://www.bitgo.com/api/v2/#estimate-transaction-fees) has an API but it requires authentication. Earn.com provides estimates anonymously and they are backed by Coinbase so I feel confident that it'll remain online for the near future. 

LND doesn't provide any details on fee estimation until after the transaction is sent. They use the underlying btcd/bitcoind RPC to get the fee. I have a thought to submit a PR to add a new RPC endpoint to calculate the fee of a transaction without sending it. This way, we can display the exact number of satoshis to the user and we wouldn't need to use the earn API at all. I know this would likely take a long time to get implemented, approved, merged, released, then installed by users. So I figured to go with the earn API for now.

I also looked at a few other lightning wallets to see what they use:
1. Lightning App - no fee estimation, always displays 0 fee
2. Zap - uses earn.com ([ref](https://github.com/LN-Zap/zap-desktop/blob/master/app/lib/utils/api.js#L42))
3. BlueWallet - uses earn.com ([ref](https://github.com/BlueWallet/BlueWallet/blob/master/models/networkTransactionFees.js#L17))

There is still a limitation using this API. The earn.com fee estimates are returned as sats/byte based on the current mempool. Since Joule cannot know how many bytes the tx will be ahead of time, we cannot display the exact number of satoshis the transaction will cost. So right now I just display the sats/byte. It's not ideal, but I feel like it's better than not displaying a fee at all.

Reply:
Thorough analysis, thanks. I think you made the best decision with that information 👍